packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -101,19 +102,29 @@ async function instantiateModule(
     root
   } = server.config
 
+  // The `extensions` and `mainFields` options are used to ensure that
+  // CommonJS modules are preferred. We want to avoid ESM->ESM imports
+  // whenever possible, because `hookNodeResolve` can't intercept them.
   const resolveOptions: InternalResolveOptions = {
-    conditions: ['node'],
+    // By adding "require" to the `conditions` array, resolution of the
+    // pkg.exports field will use "require" condition whenever it comes
+    // before "import" condition.
+    conditions: ['node', 'require'],
     dedupe,
-    // Prefer CommonJS modules.
-    extensions: ['.js', '.mjs', '.ts', '.jsx', '.tsx', '.json'],
+    extensions: ['.js', '.cjs', '.mjs', '.jsx', '.json'],
```

Comment:
Why `.ts` and `.tsx` where removed here?

Reply:
`tryNodeResolve` is only used for SSR externals, so uncompiled TypeScript is not supported